src/hooks/stores/useSyncStore.ts

```diff
@@ -0,0 +1,102 @@
+import { useMemo } from "react";
+import { create } from "zustand";
+
+type Progress = {
+  completed: number;
+  total: number;
+};
+
+type ActiveSync = {
+  progress: Progress;
+  syncIsFinished: boolean;
+};
+
+type AllSyncs = Record<string, ActiveSync>;
+
+type SyncStore = {
+  allSyncs: AllSyncs;
+  setIndivdualSync: (deviceId: string, progress?: Progress) => void;
+  resetSyncStore: () => void;
+};
+
+const emptyProgress: Progress = {
+  completed: 0,
+  total: 0,
+};
+
+const useSyncStore = create<SyncStore>()((set) => ({
+  allSyncs: {},
+  setIndivdualSync: (deviceId, progress) =>
+    set((state) => {
+      if (!progress) {
+        return {
+          ...state,
+          allSyncs: {
+            ...state.allSyncs,
+            [deviceId]: {
+              syncIsFinished: state.allSyncs[deviceId].syncIsFinished || false,
+              progress: state.allSyncs[deviceId].progress || emptyProgress,
+            },
+          },
+        };
+      }
+
+      return {
+        ...state,
+        allSyncs: {
+          ...state.allSyncs,
+          [deviceId]: {
+            progress,
+            syncIsFinished: progress.completed === progress.total,
+          },
+        },
+      };
+    }),
+  resetSyncStore: () =>
+    set((state) => ({
+      ...state,
+      allSyncs: {},
+    })),
+}));
+
+export function useIncompleteSyncs() {
+  const allSyncs = useSyncStore((state) => state.allSyncs);
+  return useMemo(
+    () =>
+      Object.entries(allSyncs)
+        .filter(([, values]) => !values.syncIsFinished)
+        .map(([key]) => allSyncs[key]),
+    [allSyncs]
+  );
+}
+
+/**
+ *
+ * @param deviceId
+ * @returns an array with [sync, setSync]
+ */
+export function useIndividualSync(deviceId: string) {
+  const allSyncs = useSyncStore((state) => state.allSyncs);
```

Comment:
out of curiosity, is there a downside to the following? not recommending it but just something that doesn't seem to be highlighted in their docs

```js
const thisSync = useSyncStore((state) => state.allSyncs[deviceId])
```

Reply:
No reason, but your way is cleaner

Reply:
Addressed: https://github.com/digidem/mapeo-mobile-shell/pull/11/commits/6f62ffe2bc4b0bfb3478f8cc5a242158098bb4de